### PR TITLE
first pass at using api to do better sentiment analysis, still need t…

### PR DESCRIPTION
**Pull request by tylerrasor**
…o tweak based on weighting

### CODE REVIEW

File: sentim_client.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+type emotionScore struct {
+	Anger          float32 `json:"Anger"`
+	Neutral        float32 `json:"Neutral"`
+	Others         float32 `json:"Others"`
+	Happy          float32 `json:"Happy"`
+	Joy            float32 `json:"Joy"`
+	Affection      float32 `json:"Affection"`
+	Sad            float32 `json:"Sad"`
+	Disappointment float32 `json:"Disappointment"`
```
Comment (tylerrasor) [inline sentim_client.go:28]: https://sentimllc.com/api.html?shell#tocs_emotionresponse
```json
{
  "EmotionScore": {
    "Anger": 0.70523544,
    "Neutral": 0.00899112,
    "Others": 0.3163906,
    "Happy": 0.11595909,
    "Joy": 0.02580458,
    "Affection": 0.01137558,
    "Sad": 0.00235727,
    "Disappointment": 0.00223313
  },
  "Emotion": "Anger"
}
```
^^ so we get back like legit weighted scores and could theoretically define our own algorithm for `what is gritty` based on thresholds of these `[0-1]` values ¯\\\_(ツ)_/¯

File: sentim_client_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"testing"
+
+	"github.com/fitzoh/jamie_chicken"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestItDoesAThing(t *testing.T) {
+	c := jamie_chicken.NewSentimClient()
+	assert.Equal(t, c.IsGritty("wtf, I actually kind of know go now :party-dinosaur:"), true)
```
Comment (tylerrasor) [inline sentim_client_test.go:12]: yeah, no tests, sue me

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,16 @@
 module github.com/fitzoh/jamie_chicken
 
+// 1.16rc1
+go 1.16
```
Comment (fitzoh) [inline go.mod:4]: Is go 1.16 required?
This is currently deployed via google cloud functions, looks like it maxes out at 1.13: https://cloud.google.com/functions/docs/concepts/exec

File: sentim_client.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	l.SetLevel(logrus.DebugLevel)
+	return &sentimClient{
+		base_url: "https://api.sentimllc.com/",
+		logger:   l,
+	}
+}
+
+func (c *sentimClient) IsGritty(message string) bool {
+	return c.isEmotion("Anger", message)
+}
```
Comment (fitzoh) [inline sentim_client.go:55]: Is `Anger` the only emotion we should be considering?

I think we need to pull in @DotNetGeekette

File: sentim_client.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	return c.isEmotion("Disappointment", message)
+}
+
+func (c *sentimClient) isEmotion(emotion string, message string) bool {
+	emot, err := c.getEmotion(message)
+	if err != nil {
+		c.logger.Debugf("something went wrong with getting emotion from the api ¯\\_(ツ)_/¯")
+		c.logger.Errorf("I supposed you want to see the error: %s", err)
+		return false
+	}
```
Comment (fitzoh) [inline sentim_client.go:67]: maybe we could set up a slack channel for logging...

File: sentim_client_test.go [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+func TestHowAngryDoIHaveToGetToBeGritty(t *testing.T) {
+	c := jamie_chicken.NewSentimClient()
+	assert.Equal(t, c.IsGritty("the fuck do I have to say to be angry"), true)
+}
+
+func TestDisappointmentIsAThing(t *testing.T) {
+	c := jamie_chicken.NewSentimClient()
+	assert.Equal(t, c.IsDisappointed("what do you mean? the outcomes I know gives wide latitude and lets teams work independently to drive produc….. hahaha couldn’t even finish the bullshit"), false)
+}
```
Comment (DotNetGeekette) [inline sentim_client_test.go:23]: These are hilarious 😂

File: sentim_client.go (hunk 5 above)
Comment (DotNetGeekette) [reply sentim_client.go:67]: That is some responsible product ownership right there.

File: sentim_client.go (hunk 4 above)
Comment (DotNetGeekette) [reply sentim_client.go:55]: Yeah, for grit that's probably right.

Comment (DotNetGeekette) [reply sentim_client.go:55]: Joy would probably be party-dinosaur

**Review by DotNetGeekette — COMMENTED** [review]
:shipit:

File: go.mod (hunk 3 above)
Comment (tylerrasor) [reply go.mod:4]: nah, `go mod tidy` added it, will clean up later

File: sentim_client.go (hunk 5 above)
Comment (tylerrasor) [reply sentim_client.go:67]: https://github.com/fitzoh/jamie-chicken/pull/8
^^ thoughts?

**Comment by tylerrasor** [comment]
one thing I do want to bring up - feels real icky to ship every single message from `#general` off to some random ass third party that is doing... who knows what with it, so I'll definitely do a random and only ship... 1/5? 1/10? messages, something like that feel right?

File: sentim_client.go (hunk 4 above)
Comment (tylerrasor) [reply sentim_client.go:55]: we could get tricky with it too though, like
```
(anger > 0.5 || sad > 0.5 || disappointment > 0.5) && (joy < 0.2 || happy < 0.2)  == :party-dumpsterfire:
```

**Comment by fitzoh** [comment]
This is currently running on google cloud functions, probably less sketchy to have google do it
https://cloud.google.com/natural-language/pricing

**Comment by tylerrasor** [comment]
> This is currently running on google cloud functions, probably less sketchy to have google do it
> https://cloud.google.com/natural-language/pricing

oh yeah, that's _way_ more responsible
closing this for now, will revisit `soon™`

**Pull request closed without merging**